packages/amplify-provider-awscloudformation/src/graphql-resource-manager/amplify-graphql-resource-manager.ts

```diff
@@ -305,21 +320,27 @@ export class GraphQLResourceManager {
           })),
       ) as { tableName: string; stackName: string }[];
     };
-    const getAllTables = () =>
-      Object.entries(currentState.stacks)
-        .map(([name, template]) => ({
-          tableName: this.getTableNameFromTemplate(template),
-          stackName: path.basename(name, '.json'),
-        }))
-        .filter(meta => !!meta.tableName);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const getAllTables = (): any => Object.entries(currentState.stacks)
+      .map(([name, template]) => ({
+        tableName: this.getTableNameFromTemplate(template),
+        stackName: path.basename(name, '.json'),
+      }))
+      .filter(meta => !!meta.tableName);
     return this.rebuildAllTables ? getAllTables() : getTablesRequiringReplacement();
   };
 
   private getTable = (gsiChange: Diff<any, any>, proj: DiffableProject): DynamoDB.Table => {
+    if (gsiChange.path[0] === ROOT_LEVEL) {
```

Comment:
overall looks good, this change probably merits a unit test

Reply:
Currently CLI doesn't create any table on the api root template. It would be the data package which will modify the delta sync table on the api root template, but that can't be done unless this is merged. E2E test can't be added until data package changes are included.

Regarding the unit tests, I've added the helpers to do it. But it would require a refactor to make it testable with mock data.